benchmarks/file_resource_benchmarks.cpp

```diff
@@ -0,0 +1,117 @@
+//////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2016-20, Lawrence Livermore National Security, LLC and Umpire
+// project contributors. See the COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (MIT)
+//////////////////////////////////////////////////////////////////////////////
+
+#include "umpire/resource/FileMemoryResource.hpp"
+
+#include "umpire/ResourceManager.hpp"
+#include "umpire/Allocator.hpp"
+
+#include <unistd.h>
+#include <time.h>
+
+const int n = 10;
+size_t Scalar = 5;
+
+void Copy(std::size_t** A, std::size_t** C){
+    for(int i = 0; i < n; i++)
+    {
+        (*A)[i] = (*C)[i];
+    }   
+}
+
+void Scale(std::size_t** B, std::size_t** C){
+    for(int i = 0; i < n; i++)
+    {
+        (*B)[i] = (*C)[i] * Scalar;
+    }   
+}
+
+void Add(std::size_t** A, std::size_t** B, std::size_t** C){
+    for(int i = 0; i < n; i++)
+    {
+        (*C)[i] = (*A)[i] + (*B)[i];
+    }   
+}
+
+void Triad(std::size_t** A, std::size_t** B, std::size_t** C){
+    for(int i = 0; i < n; i++)
+    {
+        (*A)[i] = (*B)[i] + Scalar * (*C)[i];
+    }   
+}
+
+void Allocate(umpire::Allocator alloc, std::size_t** A, std::size_t** B, std::size_t** C){
```

Comment:
I would separate the Allocation from the Initialization.  Do the allocation in 'main' and then call a new Initialization method that contains your for loop below.  This will avoid you having to pass the variable by reference.